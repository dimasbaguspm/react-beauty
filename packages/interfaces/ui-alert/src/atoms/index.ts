export * from './alert-container';
export * from './alert-description';
export * from './alert-icon';
export * from './alert-title';
export * from './alert-dismiss-button';
