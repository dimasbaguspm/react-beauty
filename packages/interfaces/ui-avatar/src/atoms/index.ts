export * from './avatar-container';
export * from './avatar-initial';
export * from './avatar-picture';
