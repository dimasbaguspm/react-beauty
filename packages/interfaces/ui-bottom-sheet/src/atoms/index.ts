export * from './bottom-sheet-body';
export * from './bottom-sheet-container';
export * from './bottom-sheet-footer';
export * from './bottom-sheet-header';
export * from './bottom-sheet-title';
export * from './bottom-sheet-close-button';
export * from './style';
export * from './use-bottom-sheet';
