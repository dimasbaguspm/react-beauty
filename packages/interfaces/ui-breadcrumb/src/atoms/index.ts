export * from './breadcrumb-container';
export * from './breadcrumb-list';
export * from './breadcrumb-item';
export * from './breadcrumb-link';
export * from './breadcrumb-divider';
