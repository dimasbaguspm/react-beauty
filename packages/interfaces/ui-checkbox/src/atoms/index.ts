export * from './checkbox-container';
export * from './checkbox-input';
export * from './checkbox-input-label';
export * from './checkbox-label';
export * from './checkbox-group';
export * from './checkbox-group-helper-text';
export * from './checkbox-input-field';
export * from './use-checkbox';
export * from './use-checkbox-group';
