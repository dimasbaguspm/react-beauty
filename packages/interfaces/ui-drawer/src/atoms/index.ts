export * from './drawer-body';
export * from './drawer-container';
export * from './drawer-dismiss-button';
export * from './drawer-footer';
export * from './drawer-header';
export * from './drawer-title';
