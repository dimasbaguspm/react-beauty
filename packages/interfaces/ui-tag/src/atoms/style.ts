import { styled } from "@linaria/react";

const sizeSm = `
  padding-top: var(--components-tag-padding-sm-y);
  padding-bottom: var(--components-tag-padding-sm-y);
  padding-right: var(--components-tag-padding-sm-x);
  padding-left: var(--components-tag-padding-sm-x);

  &[data-has-suffix-icon="true"] {
    padding-right: var(--components-tag-padding-sm-r-icon);
  }

  &[data-has-prefix-icon="true"] {
    padding-left: var(--components-tag-padding-sm-l-icon);
  }
`;

const sizeMd = `
  padding-top: var(--components-tag-padding-md-y);
  padding-bottom: var(--components-tag-padding-md-y);
  padding-right: var(--components-tag-padding-md-x);
  padding-left: var(--components-tag-padding-md-x);

  &[data-has-suffix-icon="true"] {
    padding-right: var(--components-tag-padding-md-r-icon);
  }

  &[data-has-prefix-icon="true"] {
    padding-left: var(--components-tag-padding-md-l-icon);
  }
`;

export const ElTag = styled.div`
  display: inline-flex;
  align-items: center;
  gap: var(--components-tag-gap);
  background-color: var(--components-tag-colors-bg);
  width: fit-content;
  color: var(--components-tag-colors-color);
  font-size: var(--components-tag-font-size);
  font-weight: var(--components-tag-font-weight-medium);
  line-height: var(--components-tag-line-height);
  border-radius: var(--components-tag-border-radius);

  &[data-is-uppercase="true"] {
    text-transform: uppercase;
  }

  &[data-has-bold-text="true"] {
    font-weight: var(--components-tag-font-weight-bold);
  }

  &[data-size="sm"] {
    ${sizeSm}
  }
  &[data-size="md"] {
    ${sizeMd}
  }
`;
