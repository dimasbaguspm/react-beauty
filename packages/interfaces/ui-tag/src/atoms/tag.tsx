import { forwardRef, HTMLAttributes, ReactNode } from 'react';

import { TagSize } from '../type';

import { ElTag } from './style';

export interface TagProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode;
  size?: TagSize;
  isUppercase?: boolean;
  hasBoldText?: boolean;
  prefixIcon?: ReactNode;
  suffixIcon?: ReactNode;
}

export const Tag = forwardRef<HTMLDivElement, TagProps>(
  (
    {
      size = 'md',
      isUppercase,
      children,
      prefixIcon,
      suffixIcon,
      hasBoldText,
      ...props
    },
    ref,
  ) => (
    <ElTag
      {...props}
      ref={ref}
      data-size={size}
      data-is-uppercase={isUppercase}
      data-has-bold-text={hasBoldText}
      data-has-suffix-icon={!!suffixIcon}
      data-has-prefix-icon={!!prefixIcon}
    >
      {Boolean(prefixIcon) && prefixIcon}
      {children}
      {Boolean(suffixIcon) && suffixIcon}
    </ElTag>
  ),
);
