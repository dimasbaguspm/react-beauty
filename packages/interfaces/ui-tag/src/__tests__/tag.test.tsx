import { render } from "@react-beauty/vitest/selector";

import { Tag } from "../atoms";

describe("Tag", () => {
  it("should render correctly", () => {
    expect(render(<Tag>Tag</Tag>).asFragment()).toMatchSnapshot();
  });

  it("should allow user to pass prefix and suffix icon", () => {
    expect(
      render(
        <Tag prefixIcon={<span>Prefix</span>} suffixIcon={<span>Suffix</span>}>
          Tag
        </Tag>,
      ).asFragment(),
    ).toMatchSnapshot();
  });
});
