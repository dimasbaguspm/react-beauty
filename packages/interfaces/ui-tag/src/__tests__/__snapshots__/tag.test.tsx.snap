// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag > should allow user to pass prefix and suffix icon 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0"
    data-has-prefix-icon="true"
    data-has-suffix-icon="true"
    data-size="md"
  >
    <span>
      Prefix
    </span>
    Tag
    <span>
      Suffix
    </span>
  </div>
</DocumentFragment>
`;

exports[`Tag > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0"
    data-has-prefix-icon="false"
    data-has-suffix-icon="false"
    data-size="md"
  >
    Tag
  </div>
</DocumentFragment>
`;
